mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -364,7 +362,12 @@ namespace mbf_abstract_nav
             }
             // could not compute a valid velocity command -> stop moving the robot
             publishZeroVelocity(); // command the robot to stop
+            cmd_vel_stamped.twist = geometry_msgs::Twist();  // overwrite whatever the plugin calculated
```

Comment:
Why not still exposes the controller's intention? The controller could say that the path is blocked but still generate a reference to denote its intention?

Reply:
Well, it would be a lie! last_cmd_vel informs about the last command send to the robot, and in case of error, it will be zero-velocity. Do you think makes more sense to provide the last command generated by the plugin? :thinking: Could be... can you mention a use case when this semantic is useful?

Reply:
The controller's intention can be different from the command sent to the robot. We have a separate collision check module that updates the return value of the controller, e.g. path_blocked or success. However, the control law that the controller is using could be based for example on distance to path and angle to path. So it can still generate a reference. However, the collision check module prevents this reference of being executed.

This way you can highlight a led or speak up if someone is blocking at the front or rear of the robot. Or ask to step aside. At least denote to the used what the robot's intention 

It comes down to semantics of the `last_cmd_vel`, whether it is:
- last command sent to robot
- controller's intention

Reply:
changed